interface/forms/vitals/growthchart/chart.php

```diff
@@ -1,59 +1,29 @@
 <?php
 
 /**
- * vitals growthchart chart.php
- *
- * Description
- *  Script can create growth charts for following formats:
- *    1)png image
- *    2)pdf
- *    3)html (css for printing)
- *
- * The png image and pdf require following files in current directory:
- *  2-20yo_boys_BMI.png
- *  2-20yo_girls_BMI.png
- *  birth-24mos_boys_HC.png
- *  birth-24mos_girls_HC.png
- *
- * The html (css for printing) require the following files in current directory
- *  2-20yo_boys_BMI-1.png
- *  2-20yo_boys_BMI-2.png
- *  2-20yo_girls_BMI-1.png
- *  2-20yo_girls_BMI-2.png
- *  birth-24mos_boys_HC-1.png
- *  birth-24mos_boys_HC-2.png
- *  birth-24mos_girls_HC-1.png
- *  birth-24mos_girls_HC-2.png
- *  bluedot.gif
- *  redbox.gif
- *  reddot.gif
- *  chart.php
- *  page1.css
- *  page2.css
+ * Handles the growth chart plotting and percentiles, includes charts for patients with Down Syndrome
  *
  * @package   OpenEMR
  * @link      http://www.open-emr.org
  * @author    Brady Miller <brady.g.miller@gmail.com>
+ * @author    Daniel Pflieger <daniel@growlingflea.com> <daniel@mi-squared.com>
  * @copyright Copyright (c) 2018 Brady Miller <brady.g.miller@gmail.com>
+ * @copyright Copyright (c) 2021 Daniel Pflieger <daniel@groowlingflea.com><daniel@mi-squared.com>
  * @license   https://github.com/openemr/openemr/blob/master/LICENSE GNU General Public License 3
+ *
+ *
  */
 
-require_once("../../../../interface/globals.php");
-require_once($GLOBALS['fileroot'] . "/library/patient.inc");
-
-use OpenEMR\Common\Csrf\CsrfUtils;
+include_once("../../../../interface/globals.php");
+include_once($GLOBALS['fileroot']."/library/patient.inc");
+$chartpath = $GLOBALS['fileroot']."/interface/forms/vitals/growthchart/";
 
-if (!CsrfUtils::verifyCsrfToken($_GET["csrf_token_form"])) {
-    CsrfUtils::csrfNotVerified();
-}
-
-$chartpath = $GLOBALS['fileroot'] . "/interface/forms/vitals/growthchart/";
 $name = "";
 $pid = $_GET['pid'];
 
 if ($pid == "") {
     // no pid? no graph for you.
-    echo "<p>" . xlt('Missing PID.') . ' ' .  xlt('Please close this window.') . "</p>";
+    echo "<p>" . xl('Missing PID.', '', '', ' ') .  xl('Please close this window.') . "</p>";
```

Comment:
I see this throughout the file, we still need to escape the text via xlt because a language translation string is a potential attack vector.  @bradymiller can correct me if I'm wrong here, but I'm pretty confident where xl has been changed to xlt needs to be changed back.

Reply:
I think you mean from `xlt` to `xl` has to be changed back to `xlt` :smiley:

Reply:
Noted :)

Reply:
replaced all xl( with xlt(, pull request coming soon.